duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -100,22 +106,28 @@ public Response executeHttpRequest() throws Exception {
             + method);
       }
 
-      Request request = builder.url(url)
-          .build();
+      // finish and execute request
+      Request request = builder.headers(headers.build()).url(url).build();
+      return executeRequest(request);
+    }
 
-      // Set up client.
-      OkHttpClient httpclient = new OkHttpClient();
-      if (proxy != null) {
-        httpclient.setProxy(proxy);
+    private Response executeRequest(Request request) throws Exception {
+      Response response = httpClient.newCall(request).execute();
+      int attempts = 1;
+      while (attempts < MAX_REQUEST_ATTEMPTS && response.code() == 429) {
+        sleep(Math.min(
+          (int) (Math.pow(BACKOFF_FACTOR, (attempts - 1)) * 1000) + random.nextInt(1000),
```

Comment:
For readability, can you move the actual calculation of the backoff number into a separate method?  Along the lines of
while (blah) {
  backoff = calculateBackoff()
  sleep(backoff)
}